Synview-backend/routes/user_routes.ts

```diff
@@ -8,125 +8,192 @@ import { Session } from "https://deno.land/x/oak_sessions/mod.ts";
     });
+    context.response.status = 201;
+    context.response.body = { message: "New user created!" };
+  })
+  .post("/login", async (context) => {
+    const body = await context.request.body.json();
+    try {
+      const parsedBody = EmailLoginRequestSchema.parse(body);
+      const { password, email } = parsedBody;
+
+      if (!email || !password) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Recieved invalid login request",
+        };
+        return;
+      }
+
+      const user = await prisma.users.findFirst({
+        where: {
+          email: String(email),
+        },
+      });
+
+      if (!user) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Email doesn't exist",
+        };
+        return;
+      }
+
+      const isValidPassowrd = await bycryptVerify(password, user.passwordHash);
+
+      if (!isValidPassowrd) {
+        context.response.status = 400;
+        context.response.body = {
+          error: "Wrong password",
+        };
+        return;
+      }
+
+      const userPayload = {
+        username: user.username,
+        role: user.role,
+        id: user.user_id,
+      };
 
-    if (existingUser || existingEmail) {
-      context.response.status = 400;
+      const access_token = await createToken(getPayload(userPayload));
       context.response.body = {
-        error: "Username or Email, already taken",
+        token: access_token,
       };
-      return;
-    }
-  } catch (error) {
-    context.response.status = 500;
-    context.response.body = {
-      error: error,
-    };
-    return;
-  }
-
-  const hashedPassword = await hash(password);
-
-  await prisma.user.create({
-    data: { username: username, email: email, passwordHash: hashedPassword },
-  });
-  context.response.status = 201;
-  context.response.body = { message: "New user created!" };
-}).post("/login", async (context) => {
-  const body = await context.request.body.json();
-  try {
-    const parsedBody = EmailLoginRequestSchema.parse(body);
-    const { password, email } = parsedBody;
-
-    if (!email || !password) {
-      context.response.status = 400;
+
+      context.state.session.set("Authorization", `Bearer ${access_token}`);
+      context.response.headers.set("Authorization", `Bearer ${access_token}`);
+    } catch (e) {
+      context.response.status = 500;
       context.response.body = {
-        error: "Recieved invalid login request",
+        error: "Login went wrong" + e,
       };
       return;
     }
+  });
 
-    const user = await prisma.user.findFirst({
-      where: {
-        email: String(email),
-      },
-    });
+userRouter.use(AuthMiddleware);
 
-    if (!user) {
-      context.response.status = 400;
+userRouter
+  .get("/getPayload", (context) => {
+    const payload = getPayloadFromToken(context);
+    context.response.body = payload;
+  })
```

Comment:
JWT token payloads can be parsed on the client, the header and payload are just [base64 encoded](https://jwt.io) but not encrypted. ie - this endpoint does not need to exist

Reply:
yep